### PR TITLE
Remove `doctest` upper bounds and fix version links in changelog

### PR DESCRIPTION
**Pull request by mjgpy3**

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -1,5 +1,4 @@
-resolver:
-  url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/17/5.yaml
+resolver: lts-18.2
```
Comment (pbrisbin) [inline stack.yaml:1]: :+1:

Mentioned elsewhere, but we want our custom snapshot to be how our _apps_ get stable dependencies. We don't want to be circularly inter-depending on it throughout our self-supplied dependencies. :recycle:

File: package.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ synopsis: A non-empty comonadic list zipper
 description: A non-empty comonadic list zipper
 
 dependencies:
-  - base < 5
+  - base
```
Comment (pbrisbin) [inline package.yaml]: I think Hackage rejects and upload if you don't bound `base`. Does this project use `stack-upload-action`? If so, at what `pvp-bounds`?

**Review by pbrisbin — APPROVED** [review]
Thanks for jumping on this.

I think our more recent thinking is to upload with `--pvp-bounds lower` (we used to use `both`), so these kind of relaxations should be less common once everything moves to that.

File: package.yaml (hunk 2 above)
Comment (mjgpy3) [reply package.yaml]: Updated `release.yml` to use the new release style.

**Pull request merged**